relay/relay.go

```diff
@@ -510,7 +521,32 @@ func (r *Relay) reSetupMeta(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	err = r.meta.AddDir(uuid, nil, nil)
+
+	var newPos *mysql.Position
+	var newGset gtid.Set
+	var newUUIDSufiix int
+	if r.cfg.UUIDSuffix > 0 {
+		// if bound or rebound to a source, clear all relay log and meta
+		if err = r.PurgeRelayDir(); err != nil {
+			return err
+		}
+		r.ResetMeta()
+
+		newUUIDSufiix = r.cfg.UUIDSuffix
+		// reset the UUIDSuffix
+		r.cfg.UUIDSuffix = 0
+
+		if len(r.cfg.BinLogName) != 0 {
+			newPos = &mysql.Position{Name: r.cfg.BinLogName, Pos: binlog.MinPosition.Pos}
```

Comment:
if we use `binlog.MinPosition.Pos` to re-pull binlog from upstream, is it possible that duplicate binlog events will be consumed?

Reply:
Duplicate event will be re-pulled by not re-consumed because we have syncer checkpoint ?

Reply:
if v1.0, we use a dummy event to fill the relay log file when re-pulling. in v2.0, I think need some mechanism to handle it.

rest LGTM